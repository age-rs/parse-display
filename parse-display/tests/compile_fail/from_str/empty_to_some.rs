use parse_display::FromStr;

#[derive(FromStr)]
struct X {
    #[display(opt)]
    value: Option<u32>,
}

fn main() {}
