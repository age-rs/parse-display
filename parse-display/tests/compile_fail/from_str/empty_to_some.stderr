error: Field `value` has `opt` attribute but empty string matches `Some`, so empty string will not be parsed as `None`.

       To prevent `Some` from matching empty strings,specify a pattern that excludes empty strings using `regex`, `regex_infer`, `with`, or a format string.
 --> tests/compile_fail/from_str/empty_to_some.rs:5:5
  |
5 |     #[display(opt)]
  |     ^
